app/controllers/comments_controller.rb

```diff
@@ -1,28 +1,45 @@
 class CommentsController < ApplicationController
+  before_action :article
   def create
-    @article = Article.find(params[:article_id])
-    @comment = @article.comments.create(comment_params)
-    redirect_to article_path(@article)
+    @comment = article.comments.new(comment_params) do |comment|
+      comment.user = current_user
+    end
+    if @comment.save
+      redirect_to article_path(article)
+    else 
+      redirect_to article_path(article)
+    end
   end
 
   def edit
-    @article = Article.find(params[:article_id])
-    @comment = @article.comments.edit(comment_params)
+    @comment = Comment.find(params[:id])
```

Comment:
move this to `private` and call it from `before_action` same as what you did with `article` 👍